packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,175 @@
+    throw new CLIError(
+      'Failed to build the app: No package name found. Found errors in /src/main/AndroidManifest.xml',
+    );
+  }
+
+  let packageName = packageNameMatchArray[1];
+
+  if (!validatePackageName(packageName)) {
+    logger.warn(
+      `Invalid application's package name "${chalk.bgRed(
+        packageName,
+      )}" in 'AndroidManifest.xml'. Read guidelines for setting the package name here: ${chalk.underline.dim(
+        'https://developer.android.com/studio/build/application-id',
+      )}`,
+    );
+  }
+  return packageName;
+}
+
+/**
+ * Validates that the package name is correct
+ * @param packageName
+ */
+function validatePackageName(packageName: string) {
+  return /^[a-z][a-z0-9_]*(\.[a-z][a-z0-9_]*)+$/.test(packageName);
+}
+
+/**
+ * Pull and convert a Hermes tracing profile to Chrome tracing profile
+ * @param ctx
+ * @param dstPath
+ * @param fileName
+ * @param sourceMapPath
+ * @param raw
+ * @param generateSourceMap
+ */
+export async function downloadProfile(
+  ctx: Config,
+  dstPath: string,
+  fileName?: string,
+  sourcemapPath?: string,
+  raw?: boolean,
+  shouldGenerateSourcemap?: boolean,
+) {
+  try {
+    const packageName = getPackageName(ctx);
+
+    // if file name is not specified, pull the latest file from device
+    const file = fileName || (await getLatestFile(packageName));
+    if (!file) {
+      logger.error(
+        'There is no file in the cache/ directory. Did you record a profile from the developer menu?\n',
+      );
+      process.exit(1);
+    }
+
+    logger.info(`File to be pulled: ${file}\n`);
+
+    //if destination path is not specified, pull to the current directory
+    dstPath = dstPath || ctx.root;
+
+    if (logger.isVerbose()) {
+      logger.info('Internal commands run to pull the file:\n');
+    }
+
+    //Copy the file from device's data to sdcard, then pull the file to a temp directory
+    execSyncWithLog(`adb shell run-as ${packageName} cp cache/${file} /sdcard`);
+
+    //If --raw, pull the hermes profile to dstPath
+    if (raw) {
+      execSyncWithLog(`adb pull /sdcard/${file} ${dstPath}`);
+      logger.success(`Successfully pulled the file to ${dstPath}/${file}\n`);
+    }
+
+    //Else: transform the profile to Chrome format and pull it to dstPath
+    else {
+      const osTmpDir = os.tmpdir();
+      const tempFilePath = path.join(osTmpDir, file);
+
+      execSyncWithLog(`adb pull /sdcard/${file} ${tempFilePath}`);
+
+      //If path to source map is not given
+      if (!sourcemapPath) {
+        //Get or generate the source map
+        if (shouldGenerateSourcemap) {
+          sourcemapPath = await generateSourcemap();
+        } else {
+          sourcemapPath = await findSourcemap(ctx);
+        }
+
+        //Run without source map
+        if (!sourcemapPath) {
+          logger.warn(
+            'Cannot generate or find bundle and source map, running the transformer without source map\n',
```

Comment:
Is the `\n` at the end needed? Doesn't seem to be used in other places where we use `logger[method]`